### DISCUSSION TITLE
Slow connection & disconnection on v2.0

### DISCUSSION THREAD

**Discussion by tiykuma** [Rapptz/discord.py#7776] [Q&A]
When I use await channel.connect() method and disconnect(), it takes about a minute to connect & disconnect from the vc. What should I do?

**Comment by HigherOrderLogic** [comment]
Get better Internet.

**Reply by tiykuma** [reply]
That's not the problem. I tried my VPS and it's the same as my laptop.

**Comment by scarletcafe** [comment]
Enable logging as per the documentation so that you can see the voice lifecycle. It'll help you (and us) figure out what's wrong.

**Comment by Rapptz** [comment]
This was solved in the discord.py server.

For those who come here with the same issue, the author was changing presences in a loop like so:

```py
while True:
    await bot.change_presence(activity=discord.Game('name'))
```

This exhausted the rate limit of the gateway websocket and it had to wait a minute for the rate limit to pass. The solution to this problem is to sleep between presence changes so the gateway can still do other operations.

```py
while True:
    await bot.change_presence(activity=discord.Game('name'))
    await asyncio.sleep(60 * 60 * 12)  # 12 hours
```
